leader-election-impl/src/test/java/com/palantir/leader/proxy/AwaitingLeadershipProxyTest.java

```diff
@@ -30,6 +30,7 @@
 public class AwaitingLeadershipProxyTest {
 
     @Test
+    @SuppressWarnings("SelfEquals")
```

Comment:
I think I mentioned this before, but can't see the comment - can't we just fix the test? `proxy.equals(proxy)` is not an interesting check, whereas `proxy.equals((AwaitingLeadershipProxy) proxy)` might be.